src/server/cluster/cluster_family.h

```diff
@@ -54,14 +56,40 @@ class ClusterFamily {
   ClusterSlotMigration* AddMigration(std::string host_ip, uint16_t port,
                                      std::vector<ClusterConfig::SlotRange> slots);
 
+  uint32_t CreateMigrationSession(ConnectionContext* cntx, uint16_t port);
+
+  void Flow(CmdArgList args, ConnectionContext* cntx);
+  void Sync(CmdArgList args, ConnectionContext* cntx);
+
+  struct FlowInfo {
+    facade::Connection* conn = nullptr;
+  };
+
+  struct MigrationInfo {
+    MigrationInfo() = default;
+    MigrationInfo(std::uint32_t flows_num, std::string ip, uint32_t sync_id, uint16_t port)
+        : host_ip(ip), sync_id(sync_id), port(port), flows(flows_num) {
+    }
+    std::string host_ip;
+    uint32_t sync_id;
+    uint16_t port;
+    std::vector<FlowInfo> flows;
+  };
+
+  std::shared_ptr<MigrationInfo> GetMigrationInfo(uint32_t sync_id);
+
   ClusterConfig::ClusterShard GetEmulatedShardInfo(ConnectionContext* cntx) const;
 
   ServerFamily* server_family_ = nullptr;
 
-  mutable Mutex migrations_jobs_mu_;
+  mutable Mutex migration_mu_;  // guard migrations operations
   // holds all slot migrations that are currently in progress.
   std::vector<std::unique_ptr<ClusterSlotMigration>> migrations_jobs_
-      ABSL_GUARDED_BY(migrations_jobs_mu_);
+      ABSL_GUARDED_BY(migration_mu_);
+
+  uint32_t next_sync_id_ = 1;
+  using MigrationInfoMap = absl::btree_map<uint32_t, std::shared_ptr<MigrationInfo>>;
```

Comment:
why are you using btree_map?

Reply:
because in most cases we will have only one or two simultaneous migration processes, I don't see any point in using hash table in this case

Reply:
what will be the added value of btree_map over the hash_map in this case?

Reply:
just less memory, it's not critical.